frontend2/src/components/compete/ScrimmagingRecord.tsx

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import {
+  CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum,
+  type TeamPublic,
+} from "api/_autogen";
+import { useScrimmagingRecord } from "api/compete/useCompete";
+import { useEpisodeId } from "contexts/EpisodeContext";
+import WinLossTie from "./WinLossTie";
+import Spinner from "components/Spinner";
+
+interface ScrimmagingRecordProps {
+  team: Pick<TeamPublic, "id" | "name" | "profile">;
+  hideTeamName?: boolean;
+  hideAllScrimmages?: boolean;
+  hideUnranked?: boolean;
+  hideRanked?: boolean;
+  className?: string;
+}
+
+const ScrimmagingRecord: React.FC<ScrimmagingRecordProps> = ({
+  team,
+  hideTeamName = false,
+  hideAllScrimmages = false,
+  hideUnranked = false,
+  hideRanked = false,
+  className = "",
+}) => {
+  const { episodeId } = useEpisodeId();
+  const scrimmagingRecordAll = useScrimmagingRecord({
+    episodeId,
+    teamId: team.id,
+    scrimmageType: CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum.All,
+  });
+  const scrimmagingRecordUnranked = useScrimmagingRecord({
+    episodeId,
+    teamId: team.id,
+    scrimmageType:
+      CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum.Unranked,
+  });
+  const scrimmagingRecordRanked = useScrimmagingRecord({
+    episodeId,
+    teamId: team.id,
+    scrimmageType:
+      CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum.Ranked,
+  });
+
+  return (
+    <div
+      className={`flex h-full w-full flex-col items-start gap-1 ${className}`}
+    >
+      {!hideTeamName && (
+        <div className="flex flex-1 flex-row items-center justify-start gap-4">
+          <img
+            className="h-20 w-20 rounded-full bg-gray-400 md:h-16 md:w-16"
+            src={team.profile?.avatar_url}
+          />
+          <div className="text-xl font-semibold">{team.name}</div>
+        </div>
+      )}
+      {!scrimmagingRecordAll.isSuccess ||
+      !scrimmagingRecordUnranked.isSuccess ||
+      !scrimmagingRecordRanked.isSuccess ? (
+        <div className="mb-2 mt-4 flex w-full flex-row items-center justify-center gap-3 text-gray-400">
+          Loading... <Spinner size="lg" />
```

Comment:
If fetching the winloss record was unsuccessful, we should display something like "unable to retrieve win loss data" instead of the loading component

Reply:
Did this, and added spinners in place of the data when it is loading :sunglasses: